iStore/NetworkLayer/ProductDetailsViewModel.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  ProductDetailsViewModel.swift
+//  iStore
+//
+//  Created by Mohamed Sayed on 14.02.24.
+//
+
+import Foundation
+
+final class ProductDetailsViewModel: ObservableObject {
+    
+    private let networkManager: NetworkManager
+    
+    @Published var product: Product?
+    @Published var isloading = true
+    
+    private var skip = 0
+    private let totalNumberOfPosts = 150
+    
+    
+    init(networkManager: NetworkManager) {
+        self.networkManager = networkManager
+    }
+    
+    func getProduct(with id: Int, completion: @escaping () -> Void) {
+        
+        let url = singleProduct + "\(id)"
+        
+        networkManager.loadData(with: url, skip: skip) { [weak self] (result: Result<Product,NetworkError>) in
+            
+            guard let self = self else {
+                return
+            }
+            
+            switch result {
+                case .success(let data):
+                    DispatchQueue.main.async {
+                        self.product = data
+                        self.isloading = false
+                    }
+                    completion()
+                    self.skip += 10
+                    
+                case .failure(let error):
+                    print(NetworkError.unknownError(error))
+            }
+        }
+    }
+    
+    func getProductID() -> Int {
```

Comment:
I think its better to pass non optional product as a parameter here because this method in only responsible for getting the product ID. WDYT?

Reply:
fixed that